src/geom/paint.rs

```diff
@@ -98,6 +100,24 @@ impl Color {
             Self::Cmyk(cmyk) => Self::Cmyk(cmyk.negate()),
         }
     }
+
+    /// Get a field from this color.
+    pub fn at(&self, field: &str) -> StrResult<Value> {
+        Ok(match field {
+            "hex" => self.to_rgba().to_hex().into(),
+            "rgba" => self.to_rgba().to_array().into(),
+            "cmyk" => match self {
+                Self::Cmyk(cmyk) => cmyk.to_array().into(),
+                Self::Luma(luma) => luma.to_cmyk().to_array().into(),
+                _ => Value::None, // no rgba -> cmyk conversion
```

Comment:
not sure about just returning `none` here to be honest

Reply:
Yeah, I was worried because you wouldn't be able to prevent an error if the color type were unexpected (note that I'm basically trying to avoid many usages of `repr` here)... Perhaps we can add a `color-type` method?

Reply:
^As an update to this specifically: I moved this to a method `to-cmyk()`, which now errors on the previously "None" case. In order to be able to prepare for the error, I added the `.kind` field to colors (though I might also move that to a method, but not 100% sure yet).